.github/workflows/pr.yaml

```diff
@@ -0,0 +1,16 @@
+name: PR
+on:
+  pull_request_target:
```

Comment:
From https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_target, `pull_request_target`

> is granted a read/write repository token and can access secrets, even when it is triggered from a fork

😱 

I think we'd want `pull_request` only here. From https://github.com/marketplace/actions/semantic-pull-request

> `pull_request` ... This option is viable if all contributors have write access to the repository.

Reply:
Ooh yeah 😨 I was reading from https://github.com/marketplace/actions/semantic-pull-request which doesn't mention that. Went with `pull_request_target` on the benefit of it working for PRs from forks too but will update.

Reply:
I did some more looking into this as it seemed very scary but, from what I understand, this case is actually _relatively_ safe. My understanding is that the key point is that when using the `pull_request_target` trigger, the workflow configuration from `main` is used (1). This means that, while an action that checks out the code is dangerous as anything malicious added in the PR may be executed, actions to do things like adding labels or (as in this case) validating the PR title won't execute anything (2). Unlike with the `pull_request` target which would execute the new workflow configuration (assuming it was edited in the PR), when using `pull_request_target` the changes would have to be merged before they were used. 

Having said that, in this case it seems likely the `pull_request` will be sufficient (at least for now) and therefore there's no need to change it right now.

Still not 100% so interested to see if you interpret these docs the same way?!

(1) From https://github.com/marketplace/actions/semantic-pull-request:

> This allows the action to be used in a fork-based workflow, where e.g. you want to accept pull requests in a public repository. In this case, the configuration from the main branch of your repository will be used for the check. This means that you need to have this configuration in the main branch for the action to run at all (e.g. it won't run within a PR that adds the action initially). Also if you change the configuration in a PR, the changes will not be reflected for the current PR – only subsequent ones after the changes are in the main branch.


(2) From https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_target

> This event runs in the context of the base of the pull request, rather than in the merge commit as the pull_request event does. This prevents executing unsafe workflow code from the head of the pull request that could alter your repository or steal any secrets you use in your workflow. This event allows you to do things like create workflows that label and comment on pull requests based on the contents of the event payload.


(3) https://securitylab.github.com/research/github-actions-preventing-pwn-requests (a good post on this topic)